staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -982,12 +982,18 @@ func forwardingRulesEqual(old, new *compute.ForwardingRule) bool {
 	if err != nil {
 		klog.Errorf("forwardingRulesEqual(): failed to parse resource URL from new FR, err - %v", err)
 	}
-	return (old.IPAddress == "" || new.IPAddress == "" || old.IPAddress == new.IPAddress) &&
+	resourceIDEqual := false
+	if newResourceID == nil && oldResourceID == nil {
+		resourceIDEqual = true
+	} else if newResourceID != nil && oldResourceID != nil {
+		resourceIDEqual = oldResourceID.Equal(newResourceID)
+	}
+	return resourceIDEqual &&
```

Comment:
nit, suggest moving `resourceIDEqual` to the old `oldResourceID.Equal(newResourceID) &&` line to minimize the diff

Reply:
I fixed it in https://github.com/GoogleCloudPlatform/k8s-cloud-provider/pull/59 and  updated the `vendor`.
Fixed.